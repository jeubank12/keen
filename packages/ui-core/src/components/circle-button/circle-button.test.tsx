import React from 'react';
import { render as rtlRender, fireEvent } from '@testing-library/react';
import 'jest-styled-components';

import CircleButton from './circle-button.component';

const render = (props: any = {}) => {
  const wrapper = rtlRender(<CircleButton {...props} />);

  return {
    wrapper,
    props,
  };
};

describe('@keen.io/ui-core - <CircleButton />', () => {
  test('should render CircleButton component', () => {
    const {
      wrapper: { getByTestId },
    } = render();
    const element = getByTestId('circle-button');

    expect(element).toBeInTheDocument();
  });

  test('should render CircleButton with icon', () => {
    const {
      wrapper: { getByText },
    } = render({
      icon: <i>icon</i>,
    });

    expect(getByText('icon')).toBeInTheDocument();
  });

  test('should render CircleButton secondary variant', () => {
    const {
      wrapper: { getByTestId },
    } = render({ variant: 'secondary' });
    const element = getByTestId('circle-button');

    expect(element).toMatchSnapshot();
  });

  test('should call onClick callback', () => {
    const mockFn = jest.fn();
    const {
      wrapper: { getByTestId },
    } = render({ onClick: mockFn });
    const element = getByTestId('circle-button');
    fireEvent.click(element);

    expect(mockFn).toHaveBeenCalled();
  });

  test('should not call onClick callback once button is disabled', () => {
    const mockFn = jest.fn();
    const {
      wrapper: { getByTestId },
    } = render({ onClick: mockFn, isDisabled: true });
    const element = getByTestId('circle-button');
    fireEvent.click(element);

    expect(mockFn).not.toHaveBeenCalled();
  });
});
