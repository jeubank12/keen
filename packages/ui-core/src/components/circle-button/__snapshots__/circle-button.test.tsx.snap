// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@keen.io/ui-core - <CircleButton /> should render CircleButton secondary variant 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 50%;
  padding: 0;
  width: 37px;
  height: 37px;
  outline: none;
  border: none;
  cursor: pointer;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out, border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out, -webkit-box-shadow 0.15s ease-in-out;
  transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out, border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out, -webkit-box-shadow 0.15s ease-in-out;
  background-color: #27566D;
  border-color: #27566D;
  box-shadow: 0 2px 4px 0 rgba(29,39,41,0.15);
}

.c0:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  box-shadow: 'none';
}

.c0:hover {
  background-color: #31627A;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<button
  class="c0"
  data-testid="circle-button"
  type="button"
>
  <div
    class="c1"
  />
</button>
`;
